### ISSUE TITLE
Install fails on Ubuntu 16.04 at compiling icons for Qt

### ISSUE THREAD

**Issue by samschurter** [z-classic/electrum-zcl#39]
Running pyrcc5 icons.qrc -o gui/qt/icons_rc.py
returns
```
Traceback (most recent call last)
File "/home/<user>/.local/bin/pyrcc5", line 7, in <module>
from PyQt5.pyrcc_main import main
ImportError: no module named 'PyQt5.pyrcc_main'
```

**Comment by ch4ot1c** [comment]
Try

pip3 install python3-pyqt5

**Comment by ch4ot1c** [comment]
#41

**Comment by samschurter** [comment]
python3-pyqt5 complains
```
Could not find a version that satisfies the requirement python3-pyqt5 (from versions: )         No matching distribution found for python3-pyqt5
```


 `pip3 install pyqt5` completes with no errrors, as do the apt-get commands to install python3-pyqt5 and pyqt5-dev-tools.

It's been a while since I worked with python, so I'm kinda rusty.

**Comment by SkippOtter** [comment]
python3-pyqt5 complains 
```
Could not find a version that satisfies the requirement python3-pyqt5 (from versions: )         No matching distribution found for python3-pyqt5
```
`pip3 install pyqt5` completes with no errrors, as do the apt-get commands to install python3-pyqt5 and pyqt5-dev-tools.

It's been a while since I worked with python, so I'm kinda rusty.

**Comment by SkippOtter** [comment]
python3-pyqt5 complains 
```
Could not find a version that satisfies the requirement python3-pyqt5 (from versions: )         No matching distribution found for python3-pyqt5
```
`pip3 install pyqt5` completes with no errrors, as do the apt-get commands to install python3-pyqt5 and pyqt5-dev-tools.

It's been a while since I worked with python, so I'm kinda rusty.

**Comment by ch4ot1c** [comment]
sudo apt-get update

pip install python-qt5

pip3 install --user pyqt5  
sudo apt-get install python3-pyqt5  
sudo apt-get install pyqt5-dev-tools
sudo apt-get install qttools5-dev-tools

Any of this work for you?

**Comment by samschurter** [comment]
On installing python-qt5 with pip I get the following:
```
pip install python-qt5
Collecting python-qt5
  Downloading python-qt5-0.1.10.zip (57.5MB)
    100% |████████████████████████████████| 57.5MB 16kB/s 
No files/directories in /tmp/pip-build-B5tjHG/python-qt5/pip-egg-info (from PKG-INFO)
```
Which seems to be related to this issue on python-qt5
https://github.com/pyqt/python-qt5/issues/18#issuecomment-228978960
Indicating python-qt5 is not built for linux, at least for 2.7. however `pip3 install --user pyqt5` installs successfully.

all the other packages install successfully, but compiling the icons still fails with an import error.

**Comment by samschurter** [comment]
this is a weird mess, but doesn't seem related to electrum-zcl.  
pyrcc5 is a script in `~/.local/bin`
In that script it has `from PyQt5.pyrcc_main import main`
From the python3 console I can `import PyQt5` but `import PyQt5.pyrcc_main` says no module.
PyQt5 installed to `~/.local/lib/python3.5/site-packages/PyQt5` and in that directory is the file `pyrcc_main.py` so it's all there as far as I can tell
Not sure where to go from here.

**Comment by ch4ot1c** [comment]
Please paste:

pip --version
pip3 --version

python --version
python3 --version (3.5)

Also try just 
`pip install --upgrade pip`
`pip install pyqt5`

Thanks

**Comment by SkippOtter** [comment]
skipp@linux-desktop:~$ pip -V
pip 9.0.1 from /home/skipp/.local/lib/python2.7/site-packages (python 2.7)
skipp@linux-desktop:~$ pip3 -V
pip 8.1.1 from /usr/lib/python3/dist-packages (python 3.5)
skipp@linux-desktop:~$ python -V
Python 2.7.12
skipp@linux-desktop:~$ python3 -V
Python 3.5.2




skipp@linux-desktop:~$ pip install --upgrade pip Requirement already up-to-date: pip in ./.local/lib/python2.7/site-packages
skipp@linux-desktop:~$ pip install pyqt5         Collecting pyqt5
  Could not find a version that satisfies the requirement pyqt5 (from versions: )
No matching distribution found for pyqt5

**Comment by samschurter** [comment]
```
skipp@linux-desktop:$ pip -V
pip 9.0.1 from /home/skipp/.local/lib/python2.7/site-packages (python 2.7)
skipp@linux-desktop:$ pip3 -V
pip 8.1.1 from /usr/lib/python3/dist-packages (python 3.5)
skipp@linux-desktop:$ python -V
Python 2.7.12
skipp@linux-desktop:$ python3 -V
Python 3.5.2
```

```
skipp@linux-desktop:$ pip install --upgrade pip Requirement already up-to-date: pip in ./.local/lib/python2.7/site-packages
skipp@linux-desktop:$ pip install pyqt5         Collecting pyqt5
  Could not find a version that satisfies the requirement pyqt5 (from versions: )
No matching distribution found for pyqt5
```

And for good measure
```
skipp@linux-desktop:$ uname -a
Linux linux-desktop 4.13.0-32-generic #35~16.04.1-Ubuntu SMP Thu Jan 25 10:13:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by ch4ot1c** [comment]
Please refresh your environment and try the recently merged code

**Comment by samschurter** [comment]
Running pip with requirements.txt is successful except for
```
Could not find a version that satisfies the requirement pyqt5 (from -r requirements.txt (line 16)) (from versions: )
No matching distribution found for pyqt5 (from -r requirements.txt (line 16))
```
And compiling the icons still fails. 
I've tried uninstalling pip and all other python modules, and clearing tmp and the pip cache.

Is this maybe just a problem with my python installation?

**Comment by ch4ot1c** [comment]
Perhaps, try:
https://conda.io/miniconda.html

**Comment by KaboomFox** [comment]
Can you try pip3 install PyQt5?

I see a bug open about cases on pip.

**Comment by samschurter** [comment]
@KaboomFox still does the same thing. 

I have also tried adding the path to the PyQt5 installation directly to PYTHONPATH and I still get no module found.

@ch4ot1c I will try conda tomorrow

Does anybody else have an Ubuntu 16.04 installation where it is confirmed working? I really don't want to do a clean Ubuntu installation, but if we have confirmation that it's working somewhere else I may try that.

**Comment by samschurter** [comment]
`conda install python-qt5` and `conda install pyqt5` both fail with package not found. 

I've tried creating environments with conda and installing the packages in those environments using pip, and the icons still fail to compile.

**Comment by madmack1** [comment]
Im trying to get this working on ubuntu 16.04.3 also i get this error when trying to run. 

  File "/home/unknown/Downloads/electrum-zcl-master/lib/equihash.py", line 2, in <module>
    from pyblake2 import blake2b
ImportError: No module named 'pyblake2'

I also has all the above error which went away using pip3

**Comment by salim-b** [comment]
> Im trying to get this working on ubuntu 16.04.3 also i get this error when trying to run.
> 
> File "/home/unknown/Downloads/electrum-zcl-master/lib/equihash.py", line 2, in
> from pyblake2 import blake2b
> ImportError: No module named 'pyblake2'
> 
> I also has all the above error which went away using pip3

Try [this](https://github.com/BTCP-community/electrum-zcl/issues/47#issuecomment-362042489)

**Comment by nimbosa** [comment]
this should be properly **documented**

as pointed out by @salim-b, all the necessary steps to fix is below (also works for Ubuntu 16.04 LTS):
```
sudo pip3 install pyblake2
sudo apt-get install pyqt5-dev pyqt5-dev-tools
pyrcc5 icons.qrc -o gui/qt/icons_rc.py
```
After that `./electrum` should work as expected (at least as of Electrum-ZCL version Z!1.0.2 on Ubuntu xenial, bionic builds seems to be working as well)
___

> 
> Try [this](https://github.com/z-classic/electrum-zcl/issues/47#issuecomment-362042489)
> 
> > Im trying to get this working on ubuntu 16.04.3 also i get this error when trying to run.
> > File "/home/unknown/Downloads/electrum-zcl-master/lib/equihash.py", line 2, in
> > from pyblake2 import blake2b
> > ImportError: No module named 'pyblake2'
> > I also has all the above error which went away using pip3
>
